opal/mca/memory/patcher/configure.m4

```diff
@@ -45,7 +45,7 @@ AC_DEFUN([MCA_opal_memory_patcher_CONFIG],[
     memory_patcher_have___syscall_prototype=0
 
     AC_MSG_CHECKING([for __curbrk symbol])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([extern char *__curbrk;],[char *tmp = __curbrk;])],
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([extern char *__curbrk;],[if (*__curbrk) return 1;])],
```

Comment:
Perhaps a safer form of this test could be:

```c
printf("%p\n", __curbrk);
```

I.e., call a function that is not in this compilation unit that takes the symbol we want to test, and does something harmless (i.e., print to stdout, which won't be seen on the configure stdout).  If `__curbrk` doesn't exist, the test will fail to compile.

BTW, is there a reason we're not using `AC_CHECK_DECLS` or `AC_CHECK_FUNCS` for `__curbrk`, `__mmap`, and `__syscall`?